### ISSUE TITLE
120hz Samsung note 20 ultra doesn't work always.

### ISSUE THREAD

**Issue by SD-96** [moonlight-stream/moonlight-android#907]
Hi. love the project but i've been having a bug that seems to affect all Samsung 120hz devices(tested Samsung note 20 ultra and galaxy 20 ultra ). Using Samsung's own fps measurer it seems that 120hz keep turning off and the only way to turn it on is by changing the resolution in the app repeatedly until it shows fps above 60. And I can confirm that the Samsung game launcher fps tracker is accurate as i can see the difference in smoothness. And last the fps measurer doesn't show the phone rendering at more than ca. 90-80 fps so when it does work it wont render all the frames from the pc as shown by the built inn stats viewer.

**Comment by cgutman** [comment]
What does Moonlight's stats overlay say? Can you post a screenshot of both overlays together?

Note that you will not get 120 FPS if your PC is not rendering at 120 FPS.

**Comment by imlm** [comment]
> Hi. love the project but i've been having a bug that seems to affect all Samsung 120hz devices(tested Samsung note 20 ultra and galaxy 20 ultra ). Using Samsung's own fps measurer it seems that 120hz keep turning off and the only way to turn it on is by changing the resolution in the app repeatedly until it shows fps above 60. And I can confirm that the Samsung game launcher fps tracker is accurate as i can see the difference in smoothness. And last the fps measurer doesn't show the phone rendering at more than ca. 90-80 fps so when it does work it wont render all the frames from the pc as shown by the built inn stats viewer.

Which games did you test with? On my N20U (Exynos 990) I couldn't get anything beyond 60 fps with the display set to adaptive. I tried with FFXV, Divinity 2 Original Sin Definitive Edition, Hades and The Witcher 3. I played with VSync, frame rate caps and resolution configs on the host PC to no avail. I also used Samsung's Perf Z to measure the framerate on the phone and it was maxed out at 60 fps with the bar at 100%, i.e., the display was limited to 60hz. 

> What does Moonlight's stats overlay say? Can you post a screenshot of both overlays together?
Note that you will not get 120 FPS if your PC is not rendering at 120 FPS. 

All Moonlight stats were showing the rendering was happening at 60 fps on the host, no idea why as the settings were clearly configured to go as high as possible. On FFXV menu the in-game fps counter was actually outputting 120fps even though Moonlight's stats were 60. In actual gameplay, even when the in-game fps counter was actually in the low 40s, Moonlight's stats would still show ~60.

On my Tab S7 (SD 865+) the history was a little bit different. I managed to get 120 fps on some menu screens and what it seemed to be a locked 75 fps in actual game and this time the in-game fps counter, Moonlight's stats and Samsung Perf Z fps were all "synced" at ~75.

All this seems to point out to some quirks of Samsung's variable refresh rate display :/.
